src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -386,6 +388,45 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
             await result.ExecuteResultAsync(_pageContext);
         }
 
+        private async Task<object[]> GetArguments(HandlerMethodDescriptor handler)
+        {
+            var arguments = new object[handler.Parameters.Length];
+            var valueProvider = await GetCompositeValueProvider(_page.PageContext);
```

Comment:
`var valueProvider = await CompositeValueProvider.CreateAsync(pageContext, pageContext.ValueProviderFactories);` and delete the `GetCompositeValueProvider` method.